### PR TITLE
Fix /start/plans: Special currencies not showing coupon-discounted price

### PR DESCRIPTION
**Pull request by claudiucelfilip**
Related to Automattic/martech#3500

## Proposed Changes

* Changes allow the discountPrice to be use before into price. This is needed because coupon discount is applied after intro price.

## Why are these changes being made?

* This is needed for special discounted currencies such as INR, MXN and PHP which come with a intro offer. Applying a coupon discount on top is ignored by the frontend.

## Testing Instructions

* Set you account to one of the special currencies (INR, MXN or PHP)
* Go to `/start/plans`
* The plan prices should only include the intro offer (20%)
Screenshot 2024-10-24 at 10 14 22

* Go to `/start/plans?coupon=COUPON_CODE`
* The Plan prices should show the correct discounted prices, (20% and another 50% on top from the coupon) 
Screenshot 2024-10-25 at 09 02 21

* Prices should should match the checkout prices
Screenshot 2024-10-24 at 12 59 11

* Switching to USD should only apply the coupon discount 
Screenshot 2024-10-24 at 12 55 59

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by aneeshd16 — CHANGES_REQUESTED** [review]
Thanks for taking this up!

We will also need to adjust the billing time frame description as there is currently a mismatch:
image

**Comment by claudiucelfilip** [comment]
> We will also need to adjust the billing time frame description as there is currently a mismatc

@aneeshd16 Thanks for catching that!
I added the same approach to the timeframe as well and should show the correct price.

File: packages/plans-grid-next/src/components/shared/header-price/index.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const HeaderPrice = ( { planSlug, visibleGridPlans }: HeaderPriceProps ) => {
 					/>
 					<PlanPrice
 						currencyCode={ currencyCode }
-						rawPrice={ introOffer.rawPrice.monthly }
+						rawPrice={ discountedPrice.monthly || introOffer.rawPrice.monthly }
```
Comment (aneeshd16) [inline packages/plans-grid-next/src/components/shared/header-price/index.tsx]: Sometimes, `discountedPrice.monthly` can be 0 if a 100% coupon is applied, and the discount won't be shown. Can we change the check to something like `typeof discountedPrice?. monthly === 'number'?`. 

(Related to https://github.com/Automattic/wp-calypso/pull/94579)

**Review by aneeshd16 — COMMENTED** [review]
Thanks for the fix, I left one more comment below, which I missed adding in my previous review.

File: packages/plans-grid-next/src/components/shared/header-price/index.tsx (hunk 1 above)
Comment (claudiucelfilip) [reply packages/plans-grid-next/src/components/shared/header-price/index.tsx]: Thanks for catching that, and the suggestion!
I applied it to  both cases and now 0 should be shown for these 100% coupons.
Screenshot 2024-10-26 at 09 57 54

**Review by aneeshd16 — APPROVED** [review]
Thanks for the fix! LGTM!

**Pull request merged**